src/index.tsx

```diff
@@ -48,6 +53,94 @@ function Updaters() {
     )
 }
 
+const Input = styled.input<{ error?: boolean }>`
+    border-radius: 5px;
+    width: 100%;
+`
+const Wrapper = styled.div`
+    display: flex;
+    flex-direction: column;
+    gap: 30px;
+    padding: 10%;
+    justify-content: center;
+    align-items: center;
+`
+
+function CustomApp() {
+    const [auth, setAuth] = useState(false)
+    const PASSWORD = 'gdbetatest'
+    const PASSWORD_HASH = '$2a$10$V9DPoPvZtRpg9t23wzl5c.jYYyG5VJdJx/pvBJy61WmN/01rkSfSm'
+
+    useEffect(() => {
+        const passMatch = window.location.search.match(/\bpass=\w+,\b/)
```

Comment:
/pass=(.+?)($|&)/
a url variable ends at the end of input or when hits the & separator
this will put the password in the first matching group